src/index.js

```diff
@@ -65,44 +67,32 @@ const app = new App({
     }
     await getChannels(undefined)
 
-    async function processQueue() {
+    async function doThing(history, method, params, arr) {
+        try {
+            history = await method(params)
+        } catch (e) {
+            console.log("FAILED: " + params.channel)
+            try {
+                await app.client.conversations.join({
+                    token: process.env.TOKEN,
+                    channel: params.channel
+                })
+                history = await method(params)
+            } catch (e) {
+                arr.shift()
+                continue
```

Comment:
There's no point abstracting "doThing" -- it also breaks the code since `continue` doesn't work outside loops